src/plugin/fieldViews/RepeaterFieldView.ts

```diff
@@ -85,28 +112,101 @@ export class RepeaterFieldView extends FieldView<unknown> {
       );
       return;
     }
-    tr.replaceWith(endOfRepeaterNode, endOfRepeaterNode, newNode);
+    const nodeToAddFrom = this.node.child(index);
+    // If index supplied, add from child at given index
+    const startOfNodeToAddFrom = this.getStartOfChildNode(
+      this.node,
+      index,
+      this.getPos(),
+      this.offset
+    );
+    const positionToAddFrom = startOfNodeToAddFrom + nodeToAddFrom.nodeSize;
+    tr.insert(positionToAddFrom, newNode);
     this.outerView.dispatch(tr);
   }
 
   /**
    * Remove a child from this repeater at the given index.
+   * Do not remove if we are at the minimum threshold for number of children.
    */
-  public remove(index: number) {
+  public removeChildAt(index: number, minChildren = 0) {
```

Comment:
is there anyway for the RepeaterFieldView to hold the `minChildren` so it doesn't need to be passed in here, I can see bugs creeping in if the call site forgets to specify and it falls back to the default

Reply:
I've implemented my suggestion in https://github.com/guardian/prosemirror-elements/pull/334/commits/b2dfc0cb16addda1c1491ddc60a9654c04b88c8d - let me know what you think?